lint/src/main/java/permissions/dispatcher/OnRequestPermissionsResultCallExistDetector.java

```diff
@@ -0,0 +1,105 @@
+package permissions.dispatcher;
+
+import com.android.tools.lint.detector.api.Category;
+import com.android.tools.lint.detector.api.Detector;
+import com.android.tools.lint.detector.api.Implementation;
+import com.android.tools.lint.detector.api.Issue;
+import com.android.tools.lint.detector.api.JavaContext;
+import com.android.tools.lint.detector.api.Scope;
+import com.android.tools.lint.detector.api.Severity;
+
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+
+import lombok.ast.Annotation;
+import lombok.ast.AstVisitor;
+import lombok.ast.ClassDeclaration;
+import lombok.ast.ForwardingAstVisitor;
+import lombok.ast.MethodInvocation;
+import lombok.ast.VariableReference;
+
+public class OnRequestPermissionsResultCallExistDetector extends Detector implements Detector.JavaScanner {
+    public static final Issue ISSUE = Issue.create("NeedOnRequestPermissionsResult",
+            "Call onRequestPermissionsResult",
+            "Call onRequestPermissionsResult",
+            Category.CORRECTNESS,
+            5,
+            Severity.ERROR,
+            new Implementation(OnRequestPermissionsResultCallExistDetector.class, EnumSet.of(Scope.JAVA_FILE)));
+
+    @Override
+    public AstVisitor createJavaVisitor(JavaContext context) {
+        return new OnRequestPermissionsResultChecker(context);
+    }
+
+
+    private static class OnRequestPermissionsResultChecker extends ForwardingAstVisitor {
+        private final JavaContext context;
+        private Set<String> matchingAnnotationTypeNames;
+        private boolean hasRuntimePermissionAnnotation;
+        private boolean hasOnRequestPermissionResultCall;
+        private String generatedClassName;
+        private ClassDeclaration classDeclaration;
+
+        private OnRequestPermissionsResultChecker(JavaContext context) {
+            this.context = context;
+
+            matchingAnnotationTypeNames = new HashSet<>();
+            matchingAnnotationTypeNames.add("RuntimePermissions");
+            matchingAnnotationTypeNames.add("permissions.dispatcher.RuntimePermissions");
+        }
+
+        @Override
+        public boolean visitClassDeclaration(ClassDeclaration node) {
+            classDeclaration = node;
+            generatedClassName = node.astName() + "PermissionsDispatcher";
+            return super.visitClassDeclaration(node);
+        }
+
+        @Override
+        public boolean visitAnnotation(Annotation node) {
```

Comment:
Never mind. `@RutimePermissions` always come first. Don't need to worry about order of the method call.